### PR TITLE
Added ApiKey support, removed username and password options

### PR DESCRIPTION
**Pull request by Cherepushko**
Added ApiKey support.
Username and password options were removed.

### CODE REVIEW

**Comment by Cherepushko** [comment]
@prathapreddy123 @suyograo
ElasticsearchIO from Beam 2.31 is not compatible because it requires the following options that we're not using:
idFn indexFn routingFn typeFn docVersionFn isDeleteFn
So in this pull request I've used previous version and added apikey feature manually.

**Comment by suyograo** [comment]
> ElasticsearchIO from Beam 2.31 is not compatible because it requires the following options that we're not using:
idFn indexFn routingFn typeFn docVersionFn isDeleteFn

I am not sure why 2.31 made these required. They are functions that is executed to extract the id, index, type, etc. We already pass these directly by setting them. Very weird. 

I am ok with adding this feature on top of 2.29 until we get to 2.33 (hopefully soon)

File: v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/ElasticsearchWriteOptions.java [hunk 1]
```diff
@@ -28,17 +28,11 @@
 
   void setConnectionUrl(String connectionUrl);
 
-  @Description("Username for Elasticsearch endpoint")
+  @Description("Specify ApiKey")
```
Comment (prathapreddy123) [inline v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/options/ElasticsearchWriteOptions.java:31]: Better description would be:
```
API key for access without requiring basic authentication. Refer  https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html#security-api-create-api-key-request
```

**Review by prathapreddy123 — CHANGES_REQUESTED** [review]

**Review by prathapreddy123 — APPROVED** [review]
LGTM. @suyograo - Can you please review from elastic end

**Comment by suyograo** [comment]
LGTM from Elastic side. @prathapreddy123

**Review by prathapreddy123 — APPROVED** [review]

**Pull request merged**